src/libmeasurement_kit/net/connect.cpp

```diff
@@ -237,24 +214,13 @@ void connect(std::string address, int port,
                 return;
             }
             if (settings.find("net/ssl") != settings.end()) {
-                std::string cbp
-#ifdef MK_CA_BUNDLE
-                    = MK_CA_BUNDLE
-#endif
-                ;
+                std::string cbp = MK_CA_BUNDLE;
                 if (settings.find("net/ca_bundle_path") != settings.end()) {
                     cbp = settings.at("net/ca_bundle_path");
                 }
                 logger->debug("ca_bundle_path: '%s'", cbp.c_str());
-                ErrorOr<Var<SslContext>> ssl_context = SslContext::make(cbp);
-                if (!ssl_context) {
-                    Error err = ssl_context.as_error();
-                    bufferevent_free(r->connected_bev);
-                    callback(err, make_txp<Emitter>(
-                        timeout, r, reactor, logger));
-                    return;
-                }
-                ErrorOr<SSL *> cssl = (*ssl_context)->get_client_ssl(address);
+                ErrorOr<SSL *> cssl = libssl::Cache<>::thread_local_instance()
+                    .get_client_ssl(cbp, address, logger);
```

Comment:
Code simplified. Now we use a single thread-safe operation to construct the `SSL *`. As commented originally in #1296, we really don't even need to see the `SSL_CTX *`. So, the API is such that we don't actually see it.

Reply:
If I understand correctly this is actually the key to the solution of the bug, correct?